src/index.tsx

```diff
@@ -209,10 +211,14 @@ export default class ReactToPrint extends React.Component<IReactToPrintProps> {
     public render() {
         const {
             trigger,
+            onBeforeGetContent,
         } = this.props;
 
         return React.cloneElement(trigger(), {
-            onClick: this.handlePrint,
+            onClick: () => onBeforeGetContent ? onBeforeGetContent().then((e) => {
+                e();
```

Comment:
What is `e` here? How do you know that the user's implementation will return anything at all?

Reply:
e is the function returned by the promise.
Do you think that changing the type of `onBeforeGetContet` to
`onBeforeGetContent?: () => Promise<() => void>;` 
would be enought? That way if we had a value in _onBeforeGetContent_ we'll know that hav to be a function or the user will get a compiler issue.

Reply:
But why do we need `e` at all? The user does whatever they need to do in `onBeforeGetContent` and then we continue, we don't need to do anything with a return value

Reply:
Well, In my case I needed to update the state of a component to hide some information and this would need to be a promise to ensure the update in the component. So what I did in my component was:
onBeforeGetContent = () => Promise.resolve(() => this.setState({texto: "something"}));

how would be the correct way to do this?

Reply:
`setState` does not return anything, and `Promise.resolve` returns a Promise, not a function. A bigger issue with the above is that `react-to-print` might continue with getting the content before the `setState` has taken affect since `setState` is async.

I would change it to:

```js
onBeforeGetContent = () => {
  return new Promise((resolve, reject) => {
    this.setState({ texto: "something" }, resolve);
  });
}
```

That way you ensure that `setState` has been committed before the Promise resolves.

Reply:
I think that everything is ok in my last PR